pkg/controller/certificates/trigger/policies/gatherer_test.go

```diff
@@ -42,147 +41,100 @@ import (
+				cr("cr-unrelated-rev2", "ns-1", "cert-unrelated-uid", map[string]string{"cert-manager.io/certificate-revision": "2"}),
+				cr("cr-unrelated-rev3", "ns-1", "cert-unrelated-uid", map[string]string{"cert-manager.io/certificate-revision": "3"}),
+			}},
+			wantCurCR:  cr("cr-1-rev1", "ns-1", "cert-1-uid", map[string]string{"cert-manager.io/certificate-revision": "1"}),
+			wantNextCR: cr("cr-1-rev2", "ns-1", "cert-1-uid", map[string]string{"cert-manager.io/certificate-revision": "2"}),
+		},
+		"should error when duplicate current CRs are found": {
+			givenCert: gen.Certificate("cert-1", gen.SetCertificateNamespace("ns-1"),
+				gen.SetCertificateUID("cert-1-uid"),
+				gen.SetCertificateRevision(1),
 			),
-			wantSecret: &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: "secret-1", Namespace: "default-unit-test-ns"}},
+			builder: &testpkg.Builder{CertManagerObjects: []runtime.Object{
+				cr("cr-1-rev1a", "ns-1", "cert-1-uid", map[string]string{"cert-manager.io/certificate-revision": "1"}),
+				cr("cr-1-rev1b", "ns-1", "cert-1-uid", map[string]string{"cert-manager.io/certificate-revision": "1"}),
+			}},
+			wantErr: `multiple CertificateRequests found for the 'current' revision 1, skipping issuance until no more duplicate`,
 		},
-		"should return error when multiple certificaterequests found": {
-			givenCert: gen.Certificate("cert-1", gen.SetCertificateNamespace("default-unit-test-ns"),
-				gen.SetCertificateUID("uid-1"),
-				gen.SetCertificateSecretName("secret-1"),
+		"should error when duplicate next CRs are found": {
+			givenCert: gen.Certificate("cert-1", gen.SetCertificateNamespace("ns-1"),
+				gen.SetCertificateUID("cert-1-uid"),
 				gen.SetCertificateRevision(1),
 			),
-			builder: &testpkg.Builder{
-				KubeObjects: []runtime.Object{&corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: "secret-1"}}},
-				CertManagerObjects: []runtime.Object{
-					gen.CertificateRequest("cr-1",
-						gen.AddCertificateRequestOwnerReferences(gen.CertificateRef("cert-1", "uid-1")),
-						gen.AddCertificateRequestAnnotations(map[string]string{
-							"cert-manager.io/certificate-revision": "1",
-						}),
-					),
-					gen.CertificateRequest("cr-2",
-						gen.AddCertificateRequestOwnerReferences(gen.CertificateRef("cert-1", "uid-1")),
-						gen.AddCertificateRequestAnnotations(map[string]string{
-							"cert-manager.io/certificate-revision": "1",
-						}),
-					),
-				},
-			},
-			wantErr: "multiple CertificateRequest resources exist for the current revision, not triggering new issuance until requests have been cleaned up",
+			builder: &testpkg.Builder{CertManagerObjects: []runtime.Object{
+				cr("cr-1-rev2a", "ns-1", "cert-1-uid", map[string]string{"cert-manager.io/certificate-revision": "2"}),
+				cr("cr-1-rev2b", "ns-1", "cert-1-uid", map[string]string{"cert-manager.io/certificate-revision": "2"}),
+			}},
+			wantErr: `multiple CertificateRequests found for the 'next' revision 2, skipping issuance until no more duplicate`,
```

Comment:
My suggestions are for better messages. They may require changes to other hunks and are not designed as complete fixes :-)
```suggestion
			wantErr: `multiple CertificateRequests are found for the 'next' revision 2, skip issuance until there are no more duplicates`,
```

Reply:
Your messages definitely are better, I'll change the other chunks with this!

Reply:
Out of curiosity, why "skip" over "skipping"? In my non-native English speaker mind, "skipping" shows that the "skip" will happen until there are no more duplicates. Does "skip" convey the same information?

Reply:
I think the core problem was actually the lack of an active verb (`are`) near `no more duplicate[s]`. And my fix to that forced a specific tense which impacted the tense of _skip_.

It's possible to change the tense (`are` -> `were` for the first part) in which case you could use `skipping` instead of `is skipped`.

To your question, `skip until` does convey _continuous_ which was your concern – and was what I was trying to achieve in the word choice.

I was (and still am) having trouble understanding if these lines are purely test stage descriptors (_a la_ **jUnit**?) or if they're entirely messages from the engine to a real user. In this case, I assumed the former. If it's the latter, then changing the first `are` to `were` and switching back to `skipping` would be a good idea. -- I should have asked. I think I picked/guessed based on the general awkwardness.

Reply:
> It's possible to change the tense (are -> **were** for the first part) in which case you could use **skipping** instead of is skipped.

Woah, thank you for the grammar reminder!! You are right, I need to use `were` in order to use the progressive `skipping`

Reply:
> I was (and still am) having trouble understanding if these lines are purely "test stage descriptors" (a la jUnit?) or if they're entirely messages from the engine to a real user. If it's the latter, then changing the first are to were and switching back to skipping would be a good idea.

Those would be messages coming from the engine and addressed to the user: when looking at the logs, the user must be given a clue/context as to why a "certificate requests got deleted". But to be honest, I don't even think we should be showing anything since this "state" (duplicate certificate requests) is very short-lived (a couple of milliseconds) and it is very, VERY unlikely that this logline will lead to a better debugging session.

I'll just remove this logline entirely, there is no point in keeping it 😞 

cc @munnerz

Reply:
I added this comment:

```go
// This error feels worthless: we know that the "duplicate certificate
// requests" will be fixed almost instantanously; showing this error to
// the user is pointless since it won't even help in a debug session.
// Unfortunately, we DO have to return an error just for the purpose of
// making sure that the caller function (trigger controller, readiness
// controller) will abort their sync and retrigger a new sync, with the
// hope that the duplicate will have been removed before the next
// resync.
return Input{}, fmt.Errorf("multiple CertificateRequests were found for the 'next' revision %v, issuance is skipped until there are no more duplicates", nextCRRevision)
```